addons/survey/static/src/interactions/survey_result_chart.js

```diff
@@ -331,32 +350,28 @@ publicWidget.registry.SurveyResultChart = publicWidget.Widget.extend({
                 },
             },
         };
-    },
+    }
 
     /**
-     * Loads the chart using the provided Chart library.
+     * Adds a unicode 'check' mark if the answer's text is among the question's right answers.
      *
-     * @private
+     * @param  value
+     * @param  value.text The original text of the answer
      */
-    _loadChart: function () {
-        this.$el.css({position: 'relative'});
-        var $canvas = this.$('canvas');
-        var ctx = $canvas.get(0).getContext('2d');
-        return new Chart(ctx, this.chartConfig);
-    },
+    markIfCorrect(value) {
+        return value.text + (this.rightAnswers.indexOf(value.text) >= 0 ? " \u2713" : "");
+    }
 
     /**
-     * Adds a unicode 'check' mark if the answer's text is among the question's right answers.
-     * @private
-     * @param  {Object} value
-     * @param  {String} value.text The original text of the answer
+     * Loads the chart using the provided Chart library.
+     *
      */
-    _markIfCorrect: function (value) {
-        return `${value.text}${this.rightAnswers.indexOf(value.text) >= 0 ? " \u2713": ''}`;
-    },
-
-});
+    loadChart() {
+        this.el.style.position = "relative";
+        const canvas = this.el.querySelector("canvas");
+        const ctx = canvas.getContext("2d");
+        return new Chart(ctx, this.chartConfig);
+    }
+}
```

Comment:
i think you can add a destroy method to destroy the chart (chartjs library has a destroy method), to make sure we properly clean up everything (note that it was not done before, but it is a situation we need to gracefully handle, when we switch in edit mode, interactions are destroyed)